platform/linux-generic/arch/aarch64/odp_crypto_armv8.c

```diff
@@ -131,6 +150,178 @@ null_crypto_routine(odp_packet_t pkt ODP_UNUSED,
 	return ODP_CRYPTO_ALG_ERR_NONE;
 }
 
+static
+odp_crypto_alg_err_t aes_gcm_encrypt(odp_packet_t pkt,
+				     const odp_crypto_packet_op_param_t *param,
+				     odp_crypto_generic_session_t *session)
+{
+	armv8_cipher_state_t cs = {
+		.counter = {
+			.d = {0, 0}
+		}
+	};
+	uint8_t *iv_ptr;
+	uint64_t plaintext_bit_length = param->cipher_range.length * 8;
+	uint64_t iv_bit_length = session->p.cipher_iv.length * 8;
+	uint64_t aad_bit_length = session->p.auth_aad_len * 8;
+	uint32_t in_pos = param->cipher_range.offset;
+	uint32_t in_len = param->cipher_range.length;
+	int ret = 0;
+
+	/* Fail early if cipher_range is too large */
+	if (in_len > _ARM_CRYPTO_MAX_DATA_LENGTH) {
+		ODP_DBG("ARM Crypto: Packet size too large for requested operation\n");
+		ret = -1;
+		goto err;
+	}
+
+	if (param->cipher_iv_ptr)
+		iv_ptr = param->cipher_iv_ptr;
+	else if (session->p.cipher_iv.data)
+		iv_ptr = session->cipher.iv_data;
+	else
+		return ODP_CRYPTO_ALG_ERR_IV_INVALID;
+
+	cs.constants = &session->cc;
+
+	/* Set the counter */
+	if (armv8_aes_gcm_set_counter(iv_ptr, iv_bit_length, &cs) != 0) {
```

Comment:
I would remove the "Set the counter" comment as it does not really add any information about the following call that has set_counter in its name.

Reply:
Okay, I will remove it. Thanks!